src/storage/StorageHttpIngestHandler.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "storage/StorageHttpIngestHandler.h"
+#include <proxygen/httpserver/RequestHandler.h>
+#include <proxygen/lib/http/ProxygenErrorEnum.h>
+#include <proxygen/httpserver/ResponseBuilder.h>
+
+namespace nebula {
+namespace storage {
+
+using proxygen::HTTPMessage;
+using proxygen::HTTPMethod;
+using proxygen::ProxygenError;
+using proxygen::UpgradeProtocol;
+using proxygen::ResponseBuilder;
+
+void StorageHttpIngestHandler::init(nebula::kvstore::KVStore *kvstore) {
+    kvstore_ = kvstore;
+    CHECK_NOTNULL(kvstore_);
+}
+
+void StorageHttpIngestHandler::onRequest(std::unique_ptr<HTTPMessage> headers) noexcept {
+    if (headers->getMethod().value() != HTTPMethod::GET) {
+        // Unsupported method
+        err_ = HttpCode::E_UNSUPPORTED_METHOD;
+        return;
+    }
+
+    if (!headers->hasQueryParam("path") ||
+        !headers->hasQueryParam("space")) {
+        err_ = HttpCode::E_ILLEGAL_ARGUMENT;
+        return;
+    }
+
+    path_ = headers->getQueryParam("path");
+    space_ = headers->getIntQueryParam("space");
+}
+
+void StorageHttpIngestHandler::onBody(std::unique_ptr<folly::IOBuf>) noexcept {
+    // Do nothing, we only support GET
+}
+
+void StorageHttpIngestHandler::onEOM() noexcept {
+    switch (err_) {
+        case HttpCode::E_UNSUPPORTED_METHOD:
+            ResponseBuilder(downstream_)
+                .status(405, "Method Not Allowed")
+                .sendWithEOM();
+            return;
+        case HttpCode::E_ILLEGAL_ARGUMENT:
+            ResponseBuilder(downstream_)
+                .status(400, "Bad Request")
+                .sendWithEOM();
+            return;
+        default:
+            break;
+    }
+
+    if (ingestSSTFiles(space_, path_)) {
+        LOG(ERROR) << "SSTFile ingest successfully " << path_;
+        ResponseBuilder(downstream_)
+            .status(200, "SSTFile ingest successfully")
```

Comment:
why not define all of the magic string and code in a common file? because a lot of place use the magic string

Reply:
Good Idea. I will move the status code into `webservice/Common.h`  thanks  👍